src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -341,8 +341,6 @@ private ProjectKey AddProjectCore(ProjectSnapshotManager.Updater updater, string
 
         _logger.LogInformation($"Added project '{filePath}' with key {hostProject.Key} to project system.");
 
-        TryMigrateMiscellaneousDocumentsToProject(updater);
```

Comment:
Personally I think this method shouldn't be here, but for the record it wouldn't cause a problem if we actually removed projects when we get a RemoveProject call, rather than just zeroing it out. I can't for the life of me remember why we don't just remove them. Does anyone know?

Reply:
I have no idea. It's always seemed a wart to me.